src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/Protocol/DocumentMapping/RazorMapToDocumentEditsParams.cs

```diff
@@ -0,0 +1,20 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System;
+using System.Text.Json.Serialization;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.Razor.Protocol.DocumentMapping;
+
+internal sealed record class RazorMapToDocumentEditsParams
+{
+    [JsonPropertyName("kind")]
+    public RazorLanguageKind Kind { get; init; }
+
+    [JsonPropertyName("razorDocumentUri")]
+    public required Uri RazorDocumentUri { get; init; }
+
+    [JsonPropertyName("textEdits")]
+    public required TextChange[] TextEdits { get; init; }
```

Comment:
`TextChange` is not very LSP-y, and in fact I wouldn't be surprised if it doesn't serialize properly, as it isn't annotated for STJ. Should this be `TextEdit[]`?

If `TextChange` works, and is easier to consume, then consider renaming the property to make it clear. I think every other `TextEdits` property are of type `TextEdit`.

Reply:
I was just hitting this with manual testing :) Glad you caught it too. Changing to TextEdit